library/Faktory/JobState.hs

```diff
@@ -0,0 +1,53 @@
+module Faktory.JobState
+  ( JobState(..)
+  ) where
+
+import Faktory.Prelude
+
+import Control.Arrow ((&&&))
+import Control.Error.Util (note)
+import Data.Aeson
+import Data.HashMap.Strict (HashMap)
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Text as T
+
+data JobState
+  = JobStateUnknown
```

Comment:
I was gonna ask if it's possible to `TRACK GET` a job that's never had a `SET` but reading the docs I see that this status comes back.

The Haskeller part of me wishes that the types wouldn't allow me to `TRACK GET` a job that doesn't have a track set but the practical part of me says "just wrap the API and defer these kinds of things to Faktory's decisions."

Reply:
Well, reading `TRACK GET` on a job without every calling `TRACK SET` will actually be the common case; you'll get the knowledge of `state` moving through `enqueued`, `working`, to `success`/`failed` without anything ever calling `TRACK SET`. `TRACK SET` is useful if you want to go more finely-grained, such as `state: working, percent: 75`.

It'll actually take a little more work before `trackSet` can be useful to us, since it requires the Job Id, and (as it's currently implemented) we don't pass that into a consumer loop (which is the only reasonable place you'd want to set from).

However, your point can be equally applied to calling `TRACK GET` on a Job that wasn't enqueued with `custom: { track: 1 }`.